DQMOffline/Lumi/plugins/ZCounting.cc

```diff
@@ -0,0 +1,457 @@
+    fTrigger->fRecords[irec].hltPathName  = "";
+    fTrigger->fRecords[irec].hltPathIndex = (unsigned int)-1;
+    const std::string pattern = fTrigger->fRecords[irec].hltPattern;
+    if(edm::is_glob(pattern)) {  // handle pattern with wildcards (*,?)
+      std::vector<std::vector<std::string>::const_iterator> matches = edm::regexMatch(triggerNames.triggerNames(), pattern);
+      if(matches.empty()) {
+        edm::LogWarning("ZCounting") << "requested pattern [" << pattern << "] does not match any HLT paths" << std::endl;
+      } else {
+        BOOST_FOREACH(std::vector<std::string>::const_iterator match, matches) {
+          fTrigger->fRecords[irec].hltPathName = *match;
+        }
+      }
+    } else {  // take full HLT path name given
+      fTrigger->fRecords[irec].hltPathName = pattern;
+    }
+    // Retrieve index in trigger menu corresponding to HLT path
+    unsigned int index = triggerNames.triggerIndex(fTrigger->fRecords[irec].hltPathName);
+    if(index < result.size()) {  // check for valid index
+      fTrigger->fRecords[irec].hltPathIndex = index;
+    }
+  }
+}
+
+//--------------------------------------------------------------------------------------------------
+bool ZCounting::isMuonTrigger(const ZCountingTrigger::TTrigger &triggerMenu, const TriggerBits &hltBits)
+{
+  return triggerMenu.pass("HLT_IsoMu27_v*",hltBits);
+}
+
+//--------------------------------------------------------------------------------------------------
+bool ZCounting::isMuonTriggerObj(const ZCountingTrigger::TTrigger &triggerMenu, const TriggerObjects &hltMatchBits)
+{
+  return triggerMenu.passObj("HLT_IsoMu27_v*","hltL3crIsoL1sMu22Or25L1f0L2f10QL3f27QL3trkIsoFiltered0p07",hltMatchBits);
+}
+//--------------------------------------------------------------------------------------------------
+bool ZCounting::passMuonID(const reco::Muon& muon, const reco::Vertex& vtx, const MuonIDTypes &idType)
+{//Muon ID selection, using internal function "DataFormats/MuonReco/src/MuonSelectors.cc
+
+  if     (idType == LooseID  && muon::isLooseMuon(muon))      return true;
+  else if(idType == MediumID && muon::isMediumMuon(muon))     return true;
+  else if(idType == TightID  && muon::isTightMuon(muon, vtx)) return true;
+  else if(idType == NoneID)                                   return true;
+  else                                                        return false;
+}
+//--------------------------------------------------------------------------------------------------
+bool ZCounting::passMuonIso(const reco::Muon& muon, const MuonIsoTypes &isoType, const float isoCut)
+{//Muon isolation selection, up-to-date with MUO POG recommendation
+
+    if(isoType == TrackerIso && muon.isolationR03().sumPt < isoCut) return true;
+    else if(isoType == PFIso && muon.pfIsolationR04().sumChargedHadronPt + TMath::Max(0.,muon.pfIsolationR04().sumNeutralHadronEt + muon.pfIsolationR04().sumPhotonEt - 0.5*muon.pfIsolationR04().sumPUPt) < isoCut) return true;
```

Comment:
hi @xmniu - sorry missed this - TMath::Max can just be std::max (then you can remove all the TMath.h includes